program/src/processor.rs

```diff
@@ -6365,56 +6303,900 @@ impl Processor {
+
+        // Set SpotMarketInfo to zero
+        mango_group.spot_markets[market_index] = SpotMarketInfo {
+            spot_market: Pubkey::default(),
+            maint_asset_weight: ZERO_I80F48,
+            init_asset_weight: ZERO_I80F48,
+            maint_liab_weight: ZERO_I80F48,
+            init_liab_weight: ZERO_I80F48,
+            liquidation_fee: ZERO_I80F48,
+        };
+
+        mango_group.tokens[market_index].mint = Pubkey::default();
+        mango_group.tokens[market_index].root_bank = Pubkey::default();
+        mango_group.tokens[market_index].spot_market_mode = MarketMode::Inactive;
+        Ok(())
+    }
+
+    #[inline(never)]
+    /// For a market in `SwappingSpotMarket` where we know all the OpenOrders have been closed,
+    /// swap out the spot market for new serum spot market that uses the same base token and quote token.
+    /// Afterwards set the market back to Active
+    fn swap_spot_market(program_id: &Pubkey, accounts: &[AccountInfo]) -> MangoResult {
+        const NUM_FIXED: usize = 5;
+        let accounts = array_ref![accounts, 0, NUM_FIXED];
+
+        let [
+            mango_group_ai,     // write
+            admin_ai,           // read, signer
+            new_spot_market_ai, // read
+            old_spot_market_ai, // read
+            dex_program_ai,     // read
+        ] = accounts;
+
+        let mut mango_group = MangoGroup::load_mut_checked(mango_group_ai, program_id)?;
+        check_eq!(admin_ai.key, &mango_group.admin, MangoErrorCode::InvalidAdminKey)?;
+        check!(admin_ai.is_signer, MangoErrorCode::SignerNecessary)?;
+
+        let market_index = mango_group
+            .find_spot_market_index(old_spot_market_ai.key)
+            .ok_or(throw_err!(MangoErrorCode::InvalidMarket))?;
+
+        // Check market is in the correct state
+        check_eq!(
+            mango_group.tokens[market_index].spot_market_mode,
+            MarketMode::SwappingSpotMarket,
+            MangoErrorCode::InvalidAccountState
+        )?;
+        let new_spot_market = load_market_state(new_spot_market_ai, dex_program_ai.key)?;
+        let old_spot_market = load_market_state(old_spot_market_ai, dex_program_ai.key)?;
+        // Check base and quote mints are the same
+        check_eq!(
+            identity(new_spot_market.coin_mint),
+            identity(old_spot_market.coin_mint),
+            MangoErrorCode::Default
+        )?;
+        check_eq!(
+            identity(new_spot_market.pc_mint),
+            identity(old_spot_market.pc_mint),
+            MangoErrorCode::Default
+        )?;
+
+        mango_group.spot_markets[market_index].spot_market = *new_spot_market_ai.key;
+        mango_group.tokens[market_index].spot_market_mode = MarketMode::Active;
+
+        Ok(())
+    }
+
+    /// Only call this instruction if you know the sum of all quote positions is zero
+    /// There is no way to check it from program so must be manually checked
```

Comment:
For my understanding: Would it be correct to say, that for all accounts:
- base_position must be 0 (caused by crank via force-settle)
- quote_position must be 0 (caused by crank via settle-pnl)
- funding must be 0 (done by most instructions)
- bids/asks_quantity and taker_base/quote must be 0 (because no open orders or orders on the event queue)
- mngo_accrued must be 0 (via redeem-mango)

Reply:
I have a script to check open interest, sum of quote/base, and MNGO accrued. Will add funding and open/uncranked orders checks to it.